Automattic/wp-calypso
# Site Address Changer: Offer Site Redirect Upgrade

#### Changes proposed in this Pull Request

* Offer the Site Redirect upgrade when a user is changing their site

#### Testing instructions

Each of the options should (hopefully!) be clear. 

1. Should add the Site Redirect upgrade, change the site address, and take you immediately to the Checkout, without deleting your site - it's worth making sure the price is correct too!
2. Should change the site address without deleting the site
3. Should change the site address and delete the site (current default)

| Before                                                                                                                                                                | After                                                                                                                                                                 |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| <img width="519" alt="Screenshot 2021-09-02 at 20 08 06" src="https://user-images.githubusercontent.com/43215253/131902305-87b4d0dc-af52-4a3c-913e-9e3deb853934.png"> | <img width="644" alt="Screenshot 2021-09-02 at 19 51 35" src="https://user-images.githubusercontent.com/43215253/131902309-b52a00a5-1e60-432f-ae31-4799fcd0f264.png"> |

@supernovia, @formosattic - let me know if you have any thoughts! 

Fixes #55801

cc @jsnajdr
